### PR TITLE
Add Multiset/ECMH implementation

### PR DESCRIPTION
**Pull request by elichai**
Hi,
The first commit is cherry-picked from https://github.com/Bitcoin-ABC/secp256k1/tree/master/src/modules/multiset
and then I implemented a serliaze/parse functions and fixed wrong docs and some other minor stuff.
Best reviewed commit by commit :)

### CODE REVIEW

**Comment by elichai** [comment]
Travis found a UB in the tests: https://travis-ci.org/kaspanet/secp256k1/jobs/651495050#L391
They cast a unsigned char pointer(alignment: 1) into a uint32_t pointer(alignment 4) which is UB

> http://www.iso-9899.info/n1570.html#6.3.2.3p7 [Pointers] A pointer to an object type may be converted to a pointer to a different object type. If the resulting pointer is not correctly aligned) for the referenced type, the behavior is undefined.

(lol https://travis-ci.org/Bitcoin-ABC/secp256k1/jobs/650290389#L550)

File: include/secp256k1_multiset.h [hunk 1]
```diff
@@ -36,6 +36,32 @@ SECP256K1_API int secp256k1_multiset_init(
...
+ *  Out:       out64:  pointer to a 64-byte array to store the serialized multiset (cannot be NULL)
+ *  In:    multiset:   pointer to the multiset (cannot be NULL)
+ */
+SECP256K1_API int secp256k1_multiset_serialize(
+    const secp256k1_context* ctx,
+    unsigned char *out64,
+    const secp256k1_multiset *multiset
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Serialize a multiset to bytes
```
Comment (stasatdaglabs) [inline include/secp256k1_multiset.h]: This appears to be a copy+paste error. Please fix.

File: include/secp256k1_multiset.h [hunk 2]
```diff
@@ -36,6 +36,32 @@ SECP256K1_API int secp256k1_multiset_init(
...
+ *  Returns: 1: suceess
+ *           0: Failed to parse multiset
+ *  Args:        ctx:  pointer to a context object (cannot be NULL)
+ *  Out:    multiset:  pointer to a multiset object (cannot be NULL)
+ *  In:         in64:  pointer to the 64-byte multiset to be parsed (cannot be NULL)
+ */
+SECP256K1_API int secp256k1_multiset_parse(
+    const secp256k1_context* ctx,
+    secp256k1_multiset *multiset,
+    const unsigned char *out64
```
Comment (stasatdaglabs) [inline include/secp256k1_multiset.h]: Typo: out64 -> in64

File: src/modules/multiset/main_impl.h [hunk 3]
```diff
@@ -139,15 +135,13 @@ int secp256k1_multiset_combine(const secp256k1_context* ctx, secp256k1_multiset
...
-    secp256k1_fe_normalize(&gej_result.x);
-    secp256k1_fe_normalize(&gej_result.y);
-    secp256k1_fe_normalize(&gej_result.z);
     multiset_from_gej_var(multiset, &gej_result);
 
     return 1;
 }
 
 /** Hash the multiset into resultHash */
+/* TODO: Add hash function pointer to optionally replace the hash function */
```
Comment (stasatdaglabs) [inline src/modules/multiset/main_impl.h]: Please don't leave TODOs inside the code. Instead, create a ticket.

**Review by stasatdaglabs — CHANGES_REQUESTED** [review]

File: src/modules/multiset/main_impl.h (hunk 3 above)
Comment (elichai) [reply src/modules/multiset/main_impl.h]: Done

File: include/secp256k1_multiset.h (hunk 2 above)
Comment (elichai) [reply include/secp256k1_multiset.h]: Done

File: include/secp256k1_multiset.h (hunk 1 above)
Comment (elichai) [reply include/secp256k1_multiset.h]: Done

**Pull request merged**